static_src/stores/service_store.js

```diff
@@ -13,28 +14,34 @@ class ServiceStore extends BaseStore {
   constructor() {
     super();
     this.subscribe(() => this._registerToActions.bind(this));
-    this._data = [];
+    this._data = new Immutable.List();
   }
 
   _registerToActions(action) {
     switch (action.type) {
-      case serviceActionTypes.SERVICES_FETCH:
+      case serviceActionTypes.SERVICES_FETCH: {
         cfApi.fetchAllServices(action.orgGuid);
         break;
+      }
+
+      case serviceActionTypes.SERVICES_RECEIVED: {
+        const services = this.formatSplitResponse(action.services);
+        const immutableServices = Immutable.fromJS(services);
 
-      case serviceActionTypes.SERVICES_RECEIVED:
-        var services = this.formatSplitResponse(action.services);
-        this._data = services;
-        this.emitChange();
+        if (this.dataHasChanged(immutableServices)) {
+          this._data = immutableServices;
```

Comment:
I could see this pattern being a potential for a base store method, `replaceAll` or something.